prime-router/src/main/kotlin/azure/CheckFunction.kt

```diff
@@ -118,6 +122,78 @@ class CheckFunction : Logging {
         return HttpUtilities.httpResponse(request, responseBody.joinToString("\n") + "\n", httpStatus)
     }
 
+    /** @param code: "success" or "fail" enum for result of check's json response.
+     * want to limit to just one of the other */
+    enum class CheckResultsEnum(val code: String) { success("success"), fail("fail") }
+
+    /**
+     * Checks sftp settings works for a receiver.
+     *
+     * Differs from CheckFunction's `check()`:
+     *  - Uses OKTA oauth.
+     *  - ONLY supports POST
+     *  - ONLY does a single setting
+     *  - Does not support sending random data to a receiver
+     *  - Org and Setting names are split into two parameters
+     *  - Move CGI parameter into path
+     *  - Reply is 200 for check success and check failure cases.
+     *  - Reply is always in json since this is a REST api.
+     */
+    @FunctionName("checkreceiver")
+    fun checkreceiver(
+        @HttpTrigger(
+            name = "checkreceiver",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "checkreceiver/org/{orgName}/receiver/{receiverName}"
+        ) request: HttpRequestMessage<String?>,
+        @BindingName("orgName") orgName: String,
+        @BindingName("receiverName") receiverName: String
+    ): HttpResponseMessage {
+        val claims = AuthenticatedClaims.authenticate(request)
+
+        if (claims == null || !claims.authorized(setOf("*.*.primeadmin", "$orgName.*.admin"))) {
+            logger.warn("User '${claims?.userName}' FAILED authorized for endpoint ${request.uri}")
+            return HttpUtilities.unauthorizedResponse(request, authenticationFailure)
+        }
+
+        logger.info("User ${claims.userName} authorized for endpoint ${request.uri}")
+
+        /** Data structure to build json response */
+        data class JsonResponse(val result: CheckResultsEnum, val message: String)
+
+        val result = try {
+            val receiver =
+                BaseEngine.settingsProviderSingleton.findReceiver("$orgName.$receiverName")
+            if (receiver == null) {
+                JsonResponse(
+                    CheckResultsEnum.fail,
+                    "Org:'$orgName' Receiver Setting:'$receiverName' not found"
+                )
+            } else if (receiver.transport == null) {
+                JsonResponse(
+                    CheckResultsEnum.fail,
+                    "Org:'$orgName' Receiver Setting: '$receiverName' " +
+                        "'transport' section of setting is missing or empty - SFTP only supported"
+                )
```

Comment:
What happens here if the transport is not null but also not SFTP? Will it pass through this test?

Reply:
The test still fails, but in a less obvious way, I believe. This is logic mirrors what the old `check` endpoint does. It is similar to what happens if the settings are incorrect.

We have #7284 open to add support for other transports (e.g. REST). I'm not sure how generic that check will be because the Receiver will need some kind of API endpoint that requires credentials but allows for empty data to be submitted w/out error.